Libraries/LibWeb/Credentials/CredentialsContainer.cpp

```diff
@@ -0,0 +1,132 @@
+        return p;
+    }
+
+    // 5. If options.signal is aborted, then return a promise rejected with options.signal’s abort reason.
+    if (options.signal && options.signal->aborted()) {
+        WebIDL::reject_promise(realm(), p, options.signal->reason());
+        return p;
+    }
+
+    // 6. Let interfaces be option’s relevant credential interface objects.
+    auto const interfaces = relevant_credential_interface_objects(options);
+
+    // 7. If interfaces is empty, then return a promise rejected with a "NotSupportedError" DOMException.
+    if (interfaces.is_empty()) {
+        WebIDL::reject_promise(realm(), p, WebIDL::DOMException::create(realm(), "NotSupportedError"_string, "No credential type is supported"_string));
+        return p;
+    }
+
+    // FIXME
+    // 8. For each interface of interfaces:
+    {
+        // FIXME
+        // 1. If options.mediation is conditional and interface does not support conditional user mediation, return a promise rejected with a "TypeError" DOMException.
+        if (options.mediation == Bindings::CredentialMediationRequirement::Conditional) {
+            WebIDL::reject_promise(realm(), p, WebIDL::DOMException::create(realm(), "TypeError"_string, "Interface does not support user mediation"_string));
+            return p;
+        }
+
+        // FIXME
+        // 2. If settings’ active credential types contains interface’s [[type]], return a promise rejected with a "NotAllowedError" DOMException.
+        // 3. Append interface’s [[type]] to settings’ active credential types.
+    }
+
+    // 9. Let origin be settings’ origin.
+    auto const& origin = settings.origin();
+
+    // FIXME
+    // 10. Let sameOriginWithAncestors be true if settings is same-origin with its ancestors, and false otherwise.
+
+    // FIXME
+    // 11. For each interface in options’ relevant credential interface objects:
+    {
+        // 1. Let permission be the interface’s [[type]] Get Permissions Policy.
+        // 2. If permission is null, continue.
+        // 3. If document is not allowed to use permission, return a promise rejected with a "NotAllowedError" DOMException.
+    }
+
+    // 12. Let p be a new promise.
+
+    // FIXME
+    // 13. Run the following steps in parallel:
+    Platform::EventLoopPlugin::the().deferred_invoke(GC::create_function(vm().heap(), []() {
+    }));
+
+    // FIXME
+    // 14. React to p
+
+    // 15. Return p
+    return p;
+}
+
+CredentialsContainer::CredentialsContainer(JS::Realm& realm)
+    : PlatformObject(realm)
+{
+}
+
+void CredentialsContainer::initialize(JS::Realm& realm)
+{
+    Base::initialize(realm);
+    WEB_SET_PROTOTYPE_FOR_INTERFACE(CredentialsContainer);
+}
+
+// https://www.w3.org/TR/credential-management-1/#credentialrequestoptions-dictionary
+Vector<IDL::Interface> relevant_credential_interface_objects(Variant<CredentialCreationOptions, CredentialRequestOptions>)
```

Comment:
I doubt that we actually want to use IDL::Interface here. We only use those at codegen time. Without reading more of the spec than you've pasted here, I suspect that this will actually be a set of PlatformObjects, or perhaps some other more specific base class for all credential types.

Reply:
Changed it for Credential which is the base class for all the rest of Credential Types. This is a part of the spec which is not well defined